File: Creational/AbstractFactory/C++/products/WinButton.cpp
#include <iostream>
#include "WinButton.h"

/**
 * Concrette object of family (realisation).
 */

void WinButton::paint()
{
	std::cout << "Painting button in Windows style." << std::endl;
}

void WinButton::setPosition(const int x, const int y)
{
	std::cout << "Setting button position to (" << x << ", " << y << ") in Windows style." << std::endl;
}

File: Creational/AbstractFactory/Java/products/WinButton.java
package patterns.creational.abstractfactory.products;

import patterns.creational.abstractfactory.interfaces.IButton;

/**
 * Concrette object of family.
 */

public class WinButton implements IButton
{
	/**
	 * @Override
	 * */
	public void paint()
	{
		System.out.println("Painting button in Windows style.");
	}
	
	/**
	 * @Override
	 * */
	public void setPosition(final int x, final int y)
	{
		System.out.println("Setting button position to (" + x + ", " + y + ") in Windows style.");
	}
}
